src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -207,9 +214,26 @@ internal partial class Binder
                 }
             }
 
+            if (!hasTypeLikeConstraint && !IsNullableEnabled(typeParameterSyntax.Identifier))
+            {
```

Comment:
It would be useful to have some comments somewhere about the strategy here: we add constraints to represent the default oblivious constraint `where T: object~`, which causes us to suppress the IL emission of the default nullable constraint to imitate the behavior of pre-nullable compilers.  When reading an assembly, the absence of a constraint causes us to create the default oblivious constraint. `where T: object~`, whereas reading the default nullable constraint `where T: object?` causes us to drop it on the floor, as that is the new default.